### ISSUE TITLE
Loss in datasets with high variance in node degree.

### ISSUE THREAD

**Issue by fleverest** [deezer/fastgae#3]
Hello! Again, big fan of your work.

I just though I'd bring this to your attention: the pos_weight variable for loss in some datasets can have a very high variance when calculating it from different subgraphs. In your implementation, you calculate it once from the sample at the start, and use it for the weighted cross-entropy throughout. In a local fork, I incorporated the pos_weight as a placeholder, and updated it at each training iteration. This increased performance in my case.

I'm sorry that this is a passing comment in an issue rather than a pull request, it's just that I don't have time to write a PR right now, perhaps when the semester ends and I have some spare time?

**Comment by GuillaumeSalhaGalvan** [comment]
Dear @fleverest,

Thanks a lot for your message. This is an interesting observation. 

Indeed, the `pos_weight` is fixed in our implementation for the sake of simplicity, and as dynamic updates did not significantly impact the performances in our experiments/datasets (but I guess it depends on various factors such as the structure of your input graph, but also your sampling method, subgraph size, "sharpening" hyperparameter alpha...). 

Feel free to propose a PR when/if you have time and motivation. Meanwhile, I will keep a fixed `pos_weight` in this repo, as I want the code to stay consistent w.r.t. experiments from the paper.

Best,

Guillaume

**Comment by GuillaumeSalhaGalvan** [comment]
Also: I see that you are currently working quite a lot on the Gravity and FastGAE projects. This observation on `pos_weight` might be relevant and interesting for future users, so making it publicly visible via an issue is a good idea. 

However, in case you have more minor comments/questions, feel free to simply send me an e-mail (my mail is available [here](https://guillaumesalha.com/)) and I'll try to answer quickly.
